### PR TITLE
Upgrade Cassandra driver to 2.2.1

### PR DESCRIPTION
**Pull request by jorgebay**
Also:
- DB#_getReplication() uses metadata already loaded in the driver
- extracted module field from internal method

### CODE REVIEW

File: lib/db.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ var revPolicy = require('./revisionPolicy');
 var SchemaMigrator = require('./schemaMigration');
 var secIndexes = require('./secondaryIndexes');
 
+/** @const */
+var validTextConsistencies = {all:1, localOne: 1, localQuorum:1}
```
Comment (d00rman) [inline lib/db.js]: Missing semicolon

File: lib/db.js [hunk 2]
```diff
@@ -1111,15 +1112,17 @@ DB.prototype.getTableSchema = function(domain, table) {
...
 DB.prototype._getReplication = function(domain, table) {
     var keyspace = this._keyspaceName(domain, table);
-    var cql = "SELECT strategy_options FROM system.schema_keyspaces WHERE keyspace_name = '" + keyspace + "'";
-    return this.client.execute_p(cql, [], {consistency: this.defaultConsistency})
-    .then(function(res) {
-        var jsonOpts = res.rows.length ? res.rows[0].strategy_options : '{}';
-        var dcenters = JSON.parse(jsonOpts);
-        Object.keys(dcenters).forEach(function(dc) {
-            dcenters[dc] = parseInt(dcenters[dc]);
+    var self;
```
Comment (d00rman) [inline lib/db.js]: `var self = this;`

**Comment by d00rman** [comment]
Thank you @jorgebay ! We have in fact decided to switch to the driver version 2.2.x, cf [T114693](https://phabricator.wikimedia.org/T114693). Two minor comments in-lined.

Also, the package's version needs to be bumped.

**Comment by d00rman** [comment]
cc @wikimedia/services

**Comment by jorgebay** [comment]
Thanks! I updated my branch based on your comments.

There is still one test fail regarding keyspace replication.

File: lib/db.js [hunk 3]
```diff
@@ -1111,16 +1112,15 @@ DB.prototype.getTableSchema = function(domain, table) {
...
-            dcenters[dc] = parseInt(dcenters[dc]);
-        });
-        return dcenters;
+    var ks = this.client.metadata.keyspaces[keyspace];
+    var datacenters = {};
+    if (!ks) {
+        return datacenters;
+    }
+    Object.keys(ks.strategyOptions).forEach(function (dc) {
+        datacenters[dc] = parseInt(datacenters[dc]);
```
Comment (Pchelolo) [inline lib/db.js]: It should be `datacenters[dc] = parseInt(ks.strategyOptions[dc])` here.

**Comment by Pchelolo** [comment]
One obvious bug inline, after that's fixed - LGTM

File: package.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "version": "0.8.6",
```
Comment (Pchelolo) [inline package.json]: Please bump the version to `0.8.7`

**Comment by jorgebay** [comment]
Pushed the changes based on @Pchelolo comments

**Comment by d00rman** [comment]
Published as v0.8.7

**Pull request merged**